### PR TITLE
feat: Receive file from kmail app

### PR DESCRIPTION
**Pull request by Ambrdctr**

### CODE REVIEW

File: kDrive/AppDelegate.swift [hunk 1]
```diff
@@ -221,6 +221,19 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, AccountManagerDeleg
                 StorePresenter.showStore(from: viewController, driveFileManager: driveFileManager)
             }
             return true
+        } else if components.host == "file",
```
Comment (adrien-coye) [inline kDrive/AppDelegate.swift:224]: Nothing really, but I would prefer to use a switch on `components.host`. It's a subjective thing. As you prefer.

File: kDrive/AppDelegate.swift [hunk 2]
```diff
@@ -221,6 +221,19 @@ final class AppDelegate: UIResponder, UIApplicationDelegate, AccountManagerDeleg
                 StorePresenter.showStore(from: viewController, driveFileManager: driveFileManager)
```
Comment (adrien-coye) [inline kDrive/AppDelegate.swift:221]: All the code related to the parsing of `deeplinks` (URL components code here) should be split in a dedicated type that lives in drive-core. That way the app delegate stays clean.

Then we should think about abstracting it and sharing with drive, so we have the benefits of this for all our apps.

**Review by adrien-coye — COMMENTED** [review]
Okay, just one main thing regarding keeping code outside of the app delegate.

File: kDrive/AppDelegate.swift (hunk 1 above)
Comment (adrien-coye) [reply kDrive/AppDelegate.swift:224]: discussed, not helpful here.

File: kDrive/Utils/FileActionsHelper.swift [hunk 3]
```diff
@@ -469,7 +469,8 @@ public final class FileActionsHelper {
             floatingPanelViewController?.rightButton.isEnabled = driveFileManager.drive.accountAdmin
             floatingPanelViewController?.actionHandler = { _ in
                 driveFloatingPanelController.dismiss(animated: true) {
-                    StorePresenter.showStore(from: viewController, driveFileManager: driveFileManager)
+                    @LazyInjectService var navigationManager: NavigationManageable
```
Comment (adrien-coye) [inline kDrive/Utils/FileActionsHelper.swift:472]: It is a tiny thing really, but by convention we should only use `LazyInjectService` when used as a property.

**Comment by adrien-coye** [comment]
I did a review, it looks good to me. Philippe will give the final go.

File: kDriveCore/Utils/DeeplinkParser.swift [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import InfomaniakDI
+import SwiftUI
+
+public struct DeeplinkParser {
+    @LazyInjectService var accountManager: AccountManageable
+    @LazyInjectService var navigationManager: NavigationManageable
+
+    private let window: UIWindow?
+
+    public init(window: UIWindow?) {
```
Comment (adrien-coye) [inline kDriveCore/Utils/DeeplinkParser.swift:28]: Having a `DeeplinkParser` that requires a reference to UIKit elements seems strange.
Since we do not have an abstracted navigation, I guess we cannot change this without refactoring the navigationManager.

File: Project.swift [hunk 5]
```diff
@@ -160,7 +160,7 @@ let project = Project(name: "kDrive",
                                  bundleId: "com.infomaniak.drive.FileProvider",
                                  deploymentTarget: Constants.deploymentTarget,
                                  infoPlist: .file(path: "kDriveFileProvider/Info.plist"),
-                                 sources: "kDriveFileProvider/**",
+                                 sources: ["kDriveFileProvider/**", "kDrive/Utils/AppFactoryService.swift", "kDrive/Utils/NavigationManager.swift"],
```
Comment (PhilippeWeidmann) [inline Project.swift:163]: This looks weird, why is this needed ?

**Review by PhilippeWeidmann — DISMISSED** [review]

File: kDriveCore/Utils/DeeplinkParser.swift (hunk 4 above)
Comment (adrien-coye) [reply kDriveCore/Utils/DeeplinkParser.swift:28]: I think we can solve this with a new protocol that can provide the `rootViewController` instead of passing the window as an argument. That way the code would be more abstract, what to you think @Ambrdctr  ?

Comment (adrien-coye) [reply kDriveCore/Utils/DeeplinkParser.swift:28]: Or we should move the view related code to the `navigationManager` and pass it an enum to remove the view code from this type.

**Review by adrien-coye — COMMENTED** [review]
I did a final review. All good, only one open question about navigation code in a DeeplinkParser component.

File: kDriveCore/Utils/DeeplinkParser.swift (hunk 4 above)
Comment (adrien-coye) [reply kDriveCore/Utils/DeeplinkParser.swift]: Abstracted navigation and unit tested in last commit.

**Review by adrien-coye — APPROVED** [review]
LGTM 👍

**Review by adrien-coye — APPROVED** [review]

**Comment by adrien-coye** [comment]
Tested, all good. Merging an launching a new beta now.

**Review by adrien-coye — APPROVED** [review]
